pkg/namespace/consts.go

```diff
@@ -0,0 +1,63 @@
+package namespace
+
+import (
+	"bytes"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+)
+
+const (
+	// NamespaveVersionSize is the size of a namespace version in bytes.
+	NamespaceVersionSize = appconsts.NamespaceVersionSize
+
+	// NamespaceIDSize is the size of a namespace ID in bytes.
+	NamespaceIDSize = appconsts.NamespaceIDSize
+
+	// NamespaceSize is the size of a namespace (version + ID) in bytes.
+	NamespaceSize = appconsts.NamespaceSize
+
+	// NamespaceVersionZero is the first namespace version.
+	NamespaceVersionZero = uint8(0)
+
+	// NamespaceZeroPrefixSize is the number of `0` bytes that are prefixed to
+	// namespace IDs for version 0.
+	NamespaceVersionZeroPrefixSize = 22
+
+	// NamespaceVersionZeroIDSize is the number of bytes available for
+	// user-specified namespace ID in a namespace ID for version 0.
+	NamespaceVersionZeroIDSize = NamespaceIDSize - NamespaceVersionZeroPrefixSize
```

Comment:
[suggestion-optional] what do you think about defining a type for Namespace version `0` e.g., an array of 10 bytes? it may make type conversions and boundary checks easier.

Reply:
I'm not opposed to using a fixed length array instead of a byte slice but I don't expect it to make type conversions easier. Can you elaborate on why a fixed length array is a better choice than a slice here?

Reply:
Sure, the main advantage is that we can avoid having to check the input size in every function call to ensure that it matches the expected length. By defining and initializing the variable as an array, we can safely pass it around the code without worrying about its size.

Reply:
The namespace struct enforces an invariant for the length of a namespace so code that passes around namespaces don't need to check for length.